vdirsyncer/storage/singlefile.py

```diff
@@ -61,28 +62,14 @@ async def discover(cls, path, **kwargs):
         if kwargs.pop("collection", None) is not None:
             raise TypeError("collection argument must not be given.")
 
-        path = os.path.abspath(expand_path(path))
-        try:
-            path_glob = path % "*"
-        except TypeError:
-            # If not exactly one '%s' is present, we cannot discover
-            # collections because we wouldn't know which name to assign.
-            raise NotImplementedError
-
-        placeholder_pos = path.index("%s")
```

Comment:
See my comment here: https://github.com/pimutils/vdirsyncer/issues/1056#issuecomment-1516840449

Reply:
I think removing this block is solves your initial intent, if I understand it right (will come back with more explanation in #1056 )